docker/entrypoint.py

```diff
@@ -6,28 +6,83 @@
 import base64
 import subprocess
 import shlex
+import shutil
+from pathlib import Path
+from zipfile import ZipFile
+from typing import List
 
+SLIDES: str = "\n" + ("=" * 40) + "\n"
 
-def extract_war_files(webapps_dir, webapps_ref, path_prefix):
-    """
-    Prepare webapps on first-time startup:
-    - Extract WAR files from webapps-ref to webapps.
-    - Modify log4j to send logs to stdout.
-    - Modify context path (pathPrefix) if given.
-
-    :param webapps_dir: Path to the webapps directory.
-    :param webapps_ref: Path to the webapps reference directory.
-    :param path_prefix: Prefix for the context path.
+
+def extract(source: Path, destination: Path, prefix: str = "/", kind: str = "war"):
+    """Extract a file to a destination directory.
+
+    Args:
+        source (Path): Source directory to extract from
+        destination (Path): Destination directory to extract to
+        prefix (str, optional): Prefix extracted filepaths. Default is "/".
+        kind (str, optional): File extension to extract. Default is "war".
```

Comment:
This does not need to be a parameter; it should always be `war`. Can you think of a case where extracting a non-WAR file into webapps is needed?

Reply:
Done.